Fix chats reseting when a Sync signal comes for communities

jrainville opened the pull request:
Fixes https://github.com/status-im/status-desktop/issues/10264

The desktop issue is that the badges sometimes disappeared on both communities and chats for that community.

Turns out the issue is the SyncCommunity signal. When we receive this from the owner, we check if we are no longer in the pending state for joining, and if not, we join. However, it never checked if we are already joined, so it would re-save all the chats and that would reset all their properties to the default values.

It took me a long time to finally figure it out because it's not something I expected happened. I was more looking at the places where we actually handle the counters. Anyway, glad I could find the issue.

This also fixes the issue where `muted` would be reset.

---

FYI, the second commit is a "fix" to the CommunityImport function. At first, I thought that was the issue. That we were importing the community right before doing the store fecthes, and since we used to **not** saving the chats, we didn't register the new unread messages in the counter and then the store fetched message would just be ignored since we already imported them.

Anyway, that turned out to be a fluke, but the fix is still useful IMO and I already had created a test for that, so might as well commit it as well.

You can review per commit if you prefer.

jrainville committed:
fix(import): fix HandleImport not saving the chat

Doesn't change much, but it could have caused issues in the future, so since we might have modified the chat, we make sure to save them
Also adds a test

ghost commented:
<!--prchecklist--> 
### Pull Request Checklist
- [ ] Have you updated the documentation, if impacted (e.g. [docs.status.im](https://status.im/docs/))?
- [ ] Have you tested changes with mobile?
- [ ] Have you tested changes with desktop?

status-im-auto commented:
### Jenkins Builds
<details>
<summary>Click to see older builds (4)</summary>

| :grey_question: | Commit | :hash: | Finished (UTC) | Duration | Platform | Result |
|-|-|-|-|-|-|-|
| :heavy_check_mark: | d855605c | [#1](https://ci.status.im/job/status-go/job/prs/job/linux/job/PR-3539/1/) | 2023-05-26 17:41:01 | ~3 min | `linux` | [:package:`zip`](https://status-im-prs.ams3.cdn.digitaloceanspaces.com/status-go-desktop-230526-173757-d85560-pr3539.zip) |
| :heavy_check_mark: | d855605c | [#1](https://ci.status.im/job/status-go/job/prs/job/ios/job/PR-3539/1/) | 2023-05-26 17:42:10 | ~4 min | `ios` | [:package:`zip`](https://status-im-prs.ams3.cdn.digitaloceanspaces.com/status-go-ios-230526-173757-d85560-pr3539.zip) |
| :heavy_check_mark: | d855605c | [#1](https://ci.status.im/job/status-go/job/prs/job/android/job/PR-3539/1/) | 2023-05-26 17:42:33 | ~4 min | `android` | [:package:`aar`](https://status-im-prs.ams3.cdn.digitaloceanspaces.com/status-go-android-230526-173752-d85560-pr3539.aar) |
| :heavy_multiplication_x: | d855605c | [#1](https://ci.status.im/job/status-go/job/prs/job/tests/job/PR-3539/1/) | 2023-05-26 17:45:58 | ~8 min | `tests` | [:page_facing_up:`log`](https://ci.status.im/job/status-go/job/prs/job/tests/job/PR-3539/1//consoleText) |
</details>

| :grey_question: | Commit | :hash: | Finished (UTC) | Duration | Platform | Result |
|-|-|-|-|-|-|-|
| :heavy_check_mark: | aa542547 | [#2](https://ci.status.im/job/status-go/job/prs/job/ios/job/PR-3539/2/) | 2023-05-29 14:54:49 | ~2 min | `ios` | [:package:`zip`](https://status-im-prs.ams3.cdn.digitaloceanspaces.com/status-go-ios-230529-145212-aa5425-pr3539.zip) |
| :heavy_check_mark: | aa542547 | [#2](https://ci.status.im/job/status-go/job/prs/job/linux/job/PR-3539/2/) | 2023-05-29 14:56:57 | ~4 min | `linux` | [:package:`zip`](https://status-im-prs.ams3.cdn.digitaloceanspaces.com/status-go-desktop-230529-145212-aa5425-pr3539.zip) |
| :heavy_check_mark: | aa542547 | [#2](https://ci.status.im/job/status-go/job/prs/job/android/job/PR-3539/2/) | 2023-05-29 14:57:49 | ~5 min | `android` | [:package:`aar`](https://status-im-prs.ams3.cdn.digitaloceanspaces.com/status-go-android-230529-145208-aa5425-pr3539.aar) |
| :heavy_multiplication_x: | aa542547 | [#2](https://ci.status.im/job/status-go/job/prs/job/tests/job/PR-3539/2/) | 2023-05-29 14:58:58 | ~6 min | `tests` | [:page_facing_up:`log`](https://ci.status.im/job/status-go/job/prs/job/tests/job/PR-3539/2//consoleText) |
| | | | | | | |
| :heavy_check_mark: | f5132fc0 | [#3](https://ci.status.im/job/status-go/job/prs/job/linux/job/PR-3539/3/) | 2023-05-29 17:45:39 | ~2 min | `linux` | [:package:`zip`](https://status-im-prs.ams3.cdn.digitaloceanspaces.com/status-go-desktop-230529-174314-f5132f-pr3539.zip) |
| :heavy_check_mark: | f5132fc0 | [#3](https://ci.status.im/job/status-go/job/prs/job/ios/job/PR-3539/3/) | 2023-05-29 17:46:13 | ~2 min | `ios` | [:package:`zip`](https://status-im-prs.ams3.cdn.digitaloceanspaces.com/status-go-ios-230529-174314-f5132f-pr3539.zip) |
| :heavy_check_mark: | f5132fc0 | [#3](https://ci.status.im/job/status-go/job/prs/job/android/job/PR-3539/3/) | 2023-05-29 17:46:42 | ~3 min | `android` | [:package:`aar`](https://status-im-prs.ams3.cdn.digitaloceanspaces.com/status-go-android-230529-174309-f5132f-pr3539.aar) |
| :heavy_check_mark: | f5132fc0 | [#3](https://ci.status.im/job/status-go/job/prs/job/tests/job/PR-3539/3/) | 2023-05-29 17:55:54 | ~12 min | `tests` | [:page_facing_up:`log`](https://ci.status.im/job/status-go/job/prs/job/tests/job/PR-3539/3//consoleText) |

ilmotta reviewed (approved):
LGTM, thanks for the PR context @jrainville

caybro commented:
LGTM, tested it and it looks fine so far

osmaczko commented on protocol/messenger_communities.go:
```diff
@@ -2100,6 +2100,9 @@ func (m *Messenger) handleSyncCommunity(messageState *ReceivedMessageState, sync
 			mr, err = m.joinCommunity(context.Background(), syncCommunity.Id)
 			if err != nil {
 				logger.Debug("m.joinCommunity error", zap.Error(err))
+				if err == communities.ErrOrgAlreadyJoined {
+					return nil
+				}
 				return err
```
```suggestion
			if err != nil && err != communities.ErrOrgAlreadyJoined {
				logger.Debug("m.joinCommunity error", zap.Error(err))
				return err
```

jrainville committed:
fix(community): stop re-joining comm when receiving a sync community msg

Fixes an issue with chats being reset. Since joining a community resaves the chats with the synced default value, it resets the sate of the chats, losing the unread messages, the muted state and more.
The solution is to block the re-joining of the community. In the case of the sync, we catch that error and just continue on.

borismelnik commented on protocol/communities/manager.go:
```diff
@@ -1924,6 +1924,10 @@ func (m *Manager) JoinCommunity(id types.HexBytes) (*Community, error) {
 	}
+	if community.Joined() {
+		// Nothing to do, we are already joined
+		return community, ErrOrgAlreadyJoined
```
Based on error processing in this method, we not expect that community will be returned with error?

borismelnik replied:
Also, test not expect it, so maybe good to send `nil`?

jrainville committed:
fix tests
